src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -363,14 +363,16 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 			}
 
 			/*
-			 * If the expression is simple, such as, SELECT fn() in
-			 * PL/PgSQL code, PL engine is doing simple expression
-			 * evaluation, which can't interpret the CustomScan Node.
-			 * Function from FROM clause is not simple, so it's ok.
+			 * If the expression is simple, such as, SELECT function() or PEFORM function()
+			 * in PL/PgSQL code, PL engine does a simple expression evaluation which can't
+			 * interpret the Citus CustomScan Node.
+			 * Note: Function from FROM clause is not simple, so it's ok to pushdown.
 			 */
-			if (MaybeExecutingUDF() && IsQuerySimple(planContext->query) && !fromFuncExpr)
+			if ((MaybeExecutingUDF() || DoBlockLevel) &&
```

Comment:
Given that it's an integer I have a slight preference for `DoBlockLevel > 0`

Reply:
Chnaged